trpcultivate_phenotypes/src/Plugin/TripalImporter/TripalCultivatePhenotypesTraitsImporter.php

```diff
@@ -102,92 +137,80 @@ public static function create(ContainerInterface $container, array $configuratio
 
-    // Importer assets.
-    // All values will be accessible to every instance of the validator Plugin.
-    // This importer does not require a project and this variable is set to 0
-    // instruct validators Project + Genus that relations project-genus can be ignored.
-    $project = 0;
+    // Grab the genus from our form to use in configuring some validators
     $genus = $form_values['genus'];
-    $file_id = $form_values['file_upload'];
-
-    // Make the header columns into a simplified array where the header names
-    // are the values
-    $headers = array_keys($this->headers);
-
-    // Take our simplified headers array and flip the array keys and values
-    // This is the format that the validators will expect to know which indices
-    // in the row of data to act on
-    // For example: ['Trait Name'] => 0
-    $header_index = array_flip($headers);
 
-    // Set $skip to 0 since no validation is being done within this method
-    $skip = 0;
+    // Make the header columns into a simplified array
+    //  - Keyed by the column header name
+    //  - Values are the column header's poition in the $headers property (ie. its
+    //    index if we assume no keys were assigned)
+    $header_index = [];
+    $headers = $this->headers;
+    $i = 0;
+    foreach ($headers as $column_details) {
+      $header_index[$column_details['name']] = $i;
+      $i++;
+    }
 
     // -----------------------------------------------------
     // Metadata
-    // - Genus Exists
-    // @deprecated getValidatorIdWithScope in issue #91
-    $validator = $manager->getValidatorIdWithScope('GENUS');
-    $instance = $manager->createInstance($validator);
-    // @deprecated loadAssets in issue #93
-    $instance->loadAssets($project, $genus, $file_id, $headers, $skip);
-    // @TODO: Rename according to the new validator_id for scope 'GENUS'
-    $validators['metadata']['GENUS'] = $instance;
-
-    // - Genus matches the configured project
-    // @TODO: In a future PR, create instance for the genus-project validator
-    //        and configure it
+    // - Genus exists and is configured
+    $instance = $manager->createInstance('genus_exists');
+    $validators['metadata']['genus_exists'] = $instance;
 
     // -----------------------------------------------------
     // File level
-    // - File exists
-    // @deprecated getValidatorIdWithScope in issue #91
-    $validator = $manager->getValidatorIdWithScope('FILE');
-    $instance = $manager->createInstance($validator);
-    // @deprecated loadAssets in issue #93
-    $instance->loadAssets($project, $genus, $file_id, $headers, $skip);
-    // @TODO: Rename according to the new validator_id for scope 'FILE'
-    $validators['file']['FILE'] = $instance;
+    // - File exists and is the expected type
+    $instance = $manager->createInstance('valid_data_file');
+    // Set supported mime-types using the valid file extensions supported by this
+    // importer.
+    $supported_file_extensions = [
```

Comment:
You can pull the plugin annotaiton definition for file_types in line 27 and this way the two will always match. ie. $this->getDefinition()['file_types'];
```suggestion
    $supported_file_extensions = [
```

Reply:
This is a great suggestion! getDefinition() wasn't recognized but after a bit of digging I found that info is stored in $this->plugin_definition

This is implemented now in commit 4d5dde7